### PR TITLE
Feat(op-alloy-protocol): add deposit module 

### PR DESCRIPTION
**Pull request by DoTheBestToGetTheBest**
related https://github.com/anton-rs/kona/issues/509

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -71,7 +79,7 @@ thiserror = "1.0"
 proptest = "1.4"
 proptest-derive = "0.4"
 tokio = "1"
-
+anyhow = { version = "1.0.86", default-features = false }
```
Comment (refcell) [inline Cargo.toml:82]: Let's avoid using `anyhow`

File: crates/protocol/Cargo.toml [hunk 2]
```diff
@@ -14,25 +14,28 @@ exclude.workspace = true
...
 
+# Op-Alloy
+op-alloy-consensus.workspace = true
+
 # `serde` feature dependencies
 serde = { workspace = true, optional = true }
 alloy-serde = { workspace = true, optional = true }
 
+
+anyhow.workspace = true
```
Comment (refcell) [inline crates/protocol/Cargo.toml:34]: Remove anyhow

File: crates/protocol/src/deposits.rs [hunk 3]
```diff
@@ -0,0 +1,688 @@
...
+    /// An invalid deposit version.
+    InvalidVersion(B256),
+    /// Unexpected opaque data length
+    UnexpectedOpaqueDataLen(usize),
+    /// Failed to decode the deposit mint value.
+    MintDecode(Bytes),
+    /// Failed to decode the deposit gas value.
+    GasDecode(Bytes),
+    /// A custom error wrapping [anyhow::Error].
+    Custom(anyhow::Error),
```
Comment (refcell) [inline crates/protocol/src/deposits.rs:57]: I think we should use a generic here that implements the [`core::error::Error`](https://doc.rust-lang.org/core/error/trait.Error.html) trait.

e.g.
```rust
pub enum DepositError<T: impl core::error::Error> {
	...
	/// A custom [core::error::Error].
	Custom(T)
}
```

**Review by refcell — CHANGES_REQUESTED** [review]
Request to remove `anyhow` as a dependency.

**Comment by DoTheBestToGetTheBest** [comment]
ty for the fast review; made all changes you requested

**Review by refcell — APPROVED** [review]

**Comment by mattsse** [comment]
this did not pass CI

**Pull request merged**